### ISSUE TITLE
Improve the format registration system

### ISSUE THREAD

**Issue by rossant** [rossant/ipymd#35]
* Avoid `import *`
* Avoid using global variables
* Load modules dynamically (if possible) and use an `on_load()` function instead

**Comment by bollwyvl** [comment]
We're starting down this road over on https://github.com/jupyter/nbviewer/pull/447 and the discussion that brought me here: https://github.com/jupyter/nbviewer/issues/457

I started earlier with my own extension system inspired by some stuff I had cargo culted, but setuptools `entry_points` will do this better than I can. Here's a [good writeup](http://www.slideshare.net/doughellmann/dynamic-codepatterns). 

basically, your formats would stay, unchanged. Instead of global tomfoolery, they would be registered in `setup.py`:
```python
entry_points={
    "ipymd.format": [
      "markdown = ipymd.formats.markdown:MARKDOWN_FORMAT",
      ...
    ]
}
```

and the FormatManager would become... very small. `.load` would basically become `iter_entry_points("ipymd.format")`.

The real win is for downstream users who run ipymd _and_ ipymd extension, as they would not need to perform any additional configuration... though the received wisdom (from nose, pygments, etc.) is to still require _some_ config, but very simple, as the entry point key becomes the means of integration, not dotted paths: `--with-powerpoint`.

**Comment by rossant** [comment]
this sounds really great and much better than the current global mechanism

hopefully i'll find time to get back on this in 3-4 weeks

**Comment by rossant** [comment]
@bollwyvl let me know if you want to work on this!

**Comment by bollwyvl** [comment]
I'll give it a shot!

**Comment by rossant** [comment]
Closed by #55
